tests/src/test/java/alluxio/master/journal/ufs/JournalCheckpointThreadTest.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.journal.ufs;
+
+import alluxio.Configuration;
+import alluxio.PropertyKey;
+import alluxio.master.Master;
+import alluxio.master.journal.JournalCheckpointThread;
+import alluxio.master.journal.JournalWriter;
+import alluxio.master.journal.options.JournalWriterOptions;
+import alluxio.proto.journal.Journal;
+import alluxio.underfs.UnderFileSystem;
+import alluxio.util.CommonUtils;
+import alluxio.util.URIUtils;
+import alluxio.util.WaitForOptions;
+
+import com.google.common.base.Function;
+import org.apache.thrift.TProcessor;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.mockito.Mockito;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.ArrayDeque;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Queue;
+import java.util.Set;
+
+/**
+ * Unit tests for {@link alluxio.master.journal.JournalCheckpointThread}.
+ */
+public final class JournalCheckpointThreadTest {
+  private static class FakeMaster implements Master {
```

Comment:
I think it would be useful to move this out of this class and call it `MockMaster` ... I expect it to used in other places over time (as well as us developing the ability to override the behavior of the default implementation